scripts/initiate_ritual.py

```diff
@@ -35,36 +33,76 @@
     type=click.Choice(["GlobalAllowList", "OpenAccessAuthorizer", "ManagedAllowList"]),
     required=True,
 )
-def cli(domain, duration, network, account, access_controller):
+@click.option(
+    "--fee-model",
+    "-f",
+    help="The name of a fee model contract.",
+    type=click.Choice(["FreeFeeModel", "BqETHSubscription"]),
+    required=True,
+)
+@click.option(
+    "--num-nodes",
+    help="Number of nodes to use for the ritual.",
+    type=int,
+    required=False,
+)
+@click.option("--random-seed", help="Random seed integer for sampling.", required=False, type=int)
+@click.option("--authority", help="The address of the ritual authority.", required=False, type=str)
+@click.option(
+    "--handpicked",
+    help="The filepath of a file containing staking provider addresses.",
+    required=False,
+    type=click.Path(exists=True, readable=True),
+)
+def cli(
+    domain,
+    duration,
+    account,
+    access_controller,
+    fee_model,
+    num_nodes,
+    random_seed,
+    authority,
+    handpicked,
+):
     check_plugins()
-    print(f"Using network: {network}")
-    print(f"Using domain: {domain}")
-    print(f"Using account: {account}")
     transactor = Transactor(account=account)
-
-    if domain == LYNX:
-        providers = list(sorted(LYNX_NODES.keys()))
-    elif domain == TAPIR:
-        providers = list(sorted(TAPIR_NODES.keys()))
-    else:
-        # mainnet sampling not currently supported
-        raise ValueError(f"Sampling of providers not supported for domain '{domain}'")
-
     registry_filepath = registry_filepath_from_domain(domain=domain)
-
     chain_id = project.chain_manager.chain_id
     deployments = contracts_from_registry(filepath=registry_filepath, chain_id=chain_id)
     coordinator = deployments[project.Coordinator.contract_type.name]
 
-    access_controller = deployments[getattr(project, access_controller).contract_type.name]
-    authority = transactor.get_account().address
+    if handpicked and num_nodes:
+        raise ValueError("Cannot specify both --num-nodes and --handpicked.")
+    if not handpicked and not num_nodes:
+        raise ValueError("Must specify either --num-nodes or --handpicked.")
```

Comment:
Can we do an xor here instead?

A little tricky since num_nodes is an integer but:

```suggestion
    if not (bool(handpicked) ^ (num_nodes is not None)):
        raise ValueError(f"Specify either --num-nodes or --handpicked; got {num_nodes}, {handpicked}")
```

Reply:
✔️